### PR TITLE
update .gitattributes .gitignore

### PR DESCRIPTION
**Pull request by afragen**
Fixes https://github.com/collizo4sky/persist-admin-notices-dismissal/issues/23

Cleans up these 2 files, removing some unneeded stuff and allowing composer.json to be downloaded

### CODE REVIEW

**Comment by afragen** [comment]
Also updated `composer.json` to use classmap for autoload instead of simply including the file. It should be more efficient.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
-vendor/*
-/vendor/
-.idea/*
-index.php
+.DS_Store
+.idea
+.vs-code
```
Comment (w3guy) [inline .gitignore:3]: @afragen when developing locally, there is always a vendor/ folder hence the need to exclude it from being committed by adding it to .gitignore.

I am in favor of restoring `vendor/*`

File: .gitattributes [hunk 2]
```diff
@@ -2,4 +2,3 @@
 
 .gitattributes export-ignore
 .gitignore export-ignore
```
Comment (w3guy) [inline .gitattributes:4]: I don't know how `.gitattributes` works but I am sure you are doing it correctly.

Comment (afragen) [reply .gitattributes:4]: I am. That will make it so a distribution download will not include either of those files. I think we should include the composer.json

Comment (w3guy) [reply .gitattributes:4]: Yes include it.

File: .gitignore (hunk 1 above)
Comment (afragen) [reply .gitignore:3]: But is there a `vendor/*` folder in this project? I understand if you want it as a default for other projects. I can add it back, no harm as it won't do anything.

File: .gitattributes (hunk 2 above)
Comment (afragen) [reply .gitattributes:4]: .gitattributes works to exclude files/folders that you don't want downloaded in a distribution of the project.
.gitignore works to exclude files/folders from being in the project entirely.

File: .gitignore (hunk 1 above)
Comment (w3guy) [reply .gitignore:3]: When developing this library locally, there is always a vendor/ directory. Say you want to make a change, you clone the repo and do `composer install` which creates a vendor directory.

After making the change, if you do `git commit -am "new commit"` the vendor directory will be included. So it's good we exclude it by default.

Make sense?

Comment (afragen) [reply .gitignore:3]: Collins, that's not quite the way I work but if if makes your workflow better. Done.

**Comment by afragen** [comment]
I think we've resolved all the above issues. Anything more?

File: .gitignore (hunk 1 above)
Comment (w3guy) [reply .gitignore:3]: Curious to know how you work?

**Pull request merged**